From b21c3034680c57619ecab5104689d93198b1f531 Mon Sep 17 00:00:00 2001
From: Sean Cross <xobs@kosagi.com>
Date: Fri, 15 Jan 2016 14:00:02 +0800
Subject: [PATCH 27/39] of-thermal: support adding a hwmon device

By default, of-thermal doesn't add itself as an hwmon device.  Add
a device tree option to enable this.

Signed-off-by: Sean Cross <xobs@kosagi.com>
---
 drivers/thermal/of-thermal.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/thermal/of-thermal.c b/drivers/thermal/of-thermal.c
index d04ec3b9e5ff..b92b7e88f90d 100644
--- a/drivers/thermal/of-thermal.c
+++ b/drivers/thermal/of-thermal.c
@@ -996,6 +996,8 @@ int __init of_parse_thermal_zones(void)
 
 		/* No hwmon because there might be hwmon drivers registering */
 		tzp->no_hwmon = true;
+		if (of_property_read_bool(child, "hwmon"))
+			tzp->no_hwmon = false;
 
 		if (!of_property_read_u32(child, "sustainable-power", &prop))
 			tzp->sustainable_power = prop;
-- 
2.14.1

